### DISCUSSION TITLE
Improved Preview functionality

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#1717] [General]
Currently, when you click the preview function:

![preview](https://user-images.githubusercontent.com/1585717/131868877-4ac6f71f-4801-4cf4-bb47-44ce2896bbe6.PNG)

Vanjaro opens a new tab with the current page loaded without the admin sidebar by appending some URL parameters:

![preview_url](https://user-images.githubusercontent.com/1585717/131869099-8859ee34-fb64-4e5d-9842-9cf617920c3d.PNG)

This works OK...  But if you click on any hyperlink in the menu, the "preview" reverts back to edit mode because of the loss of the URL parameters.

It would be better if Vanjaro could offer some preview functionality similar to this:
https://themes.getbootstrap.com/preview/?theme_id=50603

![better preview](https://user-images.githubusercontent.com/1585717/131869476-7157908c-d840-46d5-8b51-60f12681160d.PNG)

Note the ability to switch the preview between Desktop and Mobile.  Also note that because the entire thing is loaded in an IFRAME, you can click around and just the IFRAME contents change.  The frame with the Desktop/Mobile switcher remains constant so the user can easily see how different pages look in both modes.

![iframe switcher](https://user-images.githubusercontent.com/1585717/131870316-236f7aa1-ef50-4c40-a1b7-1320b14ff936.PNG)

**Comment by X3Technology** [comment]
I'm not sure how to load the page in the IFRAME in such a way that the admin functionality is hidden.  You might have to open a tab in Incognito mode if that is possible.

**Comment by meetmandeep** [comment]
It would be easy to add the ability to switch between devices in preview mode. We don't need an iFrame for that. 

> But if you click on any hyperlink in the menu, the "preview" reverts back to edit mode because of the loss of the URL parameters.

This one is a bit more tricky. Do we really even need to address it? The purpose of the preview is to preview the current page. If this is indeed an issue then we can do one of the following: 

- Disable Links 
- Add preview query parameters to Internal Links
